modules/reference/pages/data-transform-rust-sdk.adoc

```diff
@@ -0,0 +1,130 @@
+= Rust SDK for Data Transforms
+:description: Work with data transforms using Rust.
+
+Rust SDK reference documentation for Redpanda data transforms.
+
+== Data transforms
+
+The `redpanda_transform_sdk` crate contains the core functions and types for transforming data within Redpanda.
+
+=== Data transforms functions
+
+==== on_record_written
+
+[,rust,role="no-copy"]
+----
+pub fn on_record_written<E, F>(cb: F) -> !
+where
+    E: Debug,
+    F: Fn(WriteEvent<'_>, &mut dyn RecordWriter) -> Result<(), E>,
+----
+
+The `on_record_written` function transforms individual records after they have been written to an input topic. The function registers a callback to be fired when a record is written to the topic. This callback is triggered after the record has been written and fsynced to disk and the producer has been acknowledged.
+
+Since this method blocks and runs forever, it should be called from the `main` function. Perform the necessary steps prior to transforming any records before calling this method.
+
+---
+
+=== Data transforms types
+
+==== Structs
+
+===== BorrowedHeader
+
+[,rust]
+----
+pub struct BorrowedHeader<'a> { /* private fields */ }
+----
+
+`BorrowedHeader` is a zero-copy <<borrowedrecord,BorrowedRecord>> header.
+
+===== BorrowedRecord
+
+[,rust]
+----
+pub struct BorrowedRecord<'a> { /* private fields */ }
+----
+
+`BorrowedRecord` is a zero-copy representation of a <<record,Record>> in Redpanda. 
+
+===== Record
+
+[,rust]
+----
+pub struct Record { /* private fields */ }
+----
+
+`Record` is a record in Redpanda. It consists of a key-value pair of bytes, along with a collection of <<recordheader,RecordHeader>>.
+
+Records are generated as the result of any transforms that act upon a `BorrowedRecord`.
+
+===== RecordHeader
+
+[,rust]
+----
+pub struct RecordHeader { /* private fields */ }
+----
+
+A `RecordHeader` is a key-value pair attached to a <<record,Record>>. Headers are opaque to the broker and are purely a mechanism for the producer and consumers to pass information.
+
+===== RecordWriter
+
+[,rust]
+----
+pub struct RecordWriter<'a> { /* private fields */ }
+----
+
+`RecordWriter` is a writer trait for output transformed records to the destination topic.
```

Comment:
Generated doc still says "trait" https://docs.rs/redpanda-transform-sdk/0.2.0/redpanda_transform_sdk/struct.RecordWriter.html Should the description be updated? Also, I'm finding "for output transformed records to the destination topic" hard to understand, can this be rephrased or is there just an incorrect word here?

Reply:
Ah thanks, yes it's a struct now, I missed that - will send out a PR.

> "for output transformed records to the destination topic" hard to understand, can this be rephrased or is there just an incorrect word here.

Yes needs rephrased: It's a struct for writing transformed records to the output topic.